### PR TITLE
Add instance id to socks metrics

### PR DESCRIPTION
**Pull request by roy-dydx**
### Changelist
Title

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
  - Introduced instance ID management functionality, improving tracking of specific instances across the application.
  - Enhanced logging and metrics tracking capabilities by incorporating instance identifiers in various components.

- **Bug Fixes**
  - Improved error handling related to instance ID retrieval and metadata requests.

- **Documentation**
  - Updated configuration schema to ensure default values for certain properties, enhancing initialization behavior.

- **Tests**
  - Added a comprehensive suite of unit tests for the instance ID module to validate its functionality and edge cases.

- **Chores**
  - Restructured build output and updated development dependencies, including the introduction of Jest for testing.

### CODE REVIEW

File: indexer/packages/base/__tests__/instance-id.test.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    });
+
+    it('should not call metadata endpoint if url is empty', async () => {
+      jest.spyOn(config, 'isStaging').mockReturnValue(true);
+      await setInstanceId();
+
+      expect(axiosRequest).not.toHaveBeenCalled();
+    });
+  });
+});
```
Comment (vincentwschau) [inline indexer/packages/base/__tests__/instance-id.test.ts:87]: nit: newline

File: indexer/packages/base/src/instance-id.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  }
+}
+
+// Exported for tests
+export function resetForTests(): void {
+  if (!config.isTest()) {
+    throw new Error(`resetForTests() cannot be called for env: ${config.NODE_ENV}`);
+  }
+  INSTANCE_ID = '';
+}
```
Comment (vincentwschau) [inline indexer/packages/base/src/instance-id.ts:50]: nit: newline

**Review by vincentwschau — APPROVED** [review]

**Review by vincentwschau — APPROVED** [review]

**Review by vincentwschau — APPROVED** [review]

**Review by vincentwschau — APPROVED** [review]

**Comment by roy-dydx** [comment]
backport release/indexer/v6.x

**Comment by roy-dydx** [comment]
backport release/indexer/v5.x

**Comment by roy-dydx** [comment]
@Mergifyio backport release/indexer/v6.x

**Comment by roy-dydx** [comment]
https://github.com/Mergifyio backport release/indexer/v5.x
